ledger/src/shred.rs

```diff
@@ -422,6 +427,16 @@ impl Shred {
         packet.meta_mut().size = size;
     }
 
+    #[cfg(any(test, feature = "dev-context-only-utils"))]
+    pub fn to_packet(&self) -> BytesPacket {
```

Comment:
What is the purpose of doing `Arc::unwrap_or_clone(Arc::clone(`  here? Seems like no matter what you do here you need to clone the contents, so you might just as well embrace the memcpy and do 
```
 let buffer: &[u8] = match self.payload() {
            Payload::Shared(bytes) => bytes.as_ref(),
            Payload::Unique(bytes) => bytes.as_ref(),
        };
        bytes::Bytes::copy_from_slice(buffer)
```

Reply:
Tbh I just copied the `Arc::unwrap_or_clone(Arc::clone(` pattern from some other places in the shred module. I can do what you're proposing. At the end, both do the same (memcopy).

Reply:
Good point. Fixed.

Reply:
thank you!